File: 1675 B. Make It Increasing/Solution.c
// 321384417
#include<stdio.h>
#include<stdbool.h>
//Krishno_Mondol_KUET_CSE
void solve(){ 
    int n; scanf("%d",&n);
    int v[n]; for(int i=0; i<n; i++){scanf("%d",&v[i]);}
    int cnt = 0; bool vd = false;
    for(int i=n-1; i>0; i--){
        if(v[i]==0){vd=true; break;}
        while(v[i-1]>=v[i]){v[i-1]/=2; cnt++;}
    }
    if(vd){printf("-1\n");}
    else{printf("%d\n",cnt);}
}
int main(){int t; scanf("%d",&t); while(t--){solve();} return 0;}

File: 1675 B. Make It Increasing/Solution.py
# 321383514
# Krishno_Mondol_KUET_CSE
def solve(): 
    n = int(input())
    v = list(map(int,input().split()))
    cnt = 0
    vd2 = False
    for i in range(n-1,0,-1):
        if v[i]==0:
            vd2 = True
            break
        while v[i-1]>=v[i]:
            v[i-1] = v[i-1]//2
            cnt += 1
    if vd2:
        print("-1")
    else:
        print(cnt)
    


t = int(input())
for i in range(t):
    solve()
